samza-core/src/main/java/org/apache/samza/storage/blobstore/BlobStoreBackupManager.java

```diff
@@ -122,6 +125,8 @@ public BlobStoreBackupManager(JobModel jobModel, ContainerModel containerModel,
     this.prevStoreSnapshotIndexesFuture = CompletableFuture.completedFuture(ImmutableMap.of());
     this.metrics = blobStoreTaskBackupMetrics;
     metrics.initStoreMetrics(storesToBackup);
+    this.blobStoreManagerMetrics = blobStoreTaskManagerMetrics;
+    blobStoreTaskManagerMetrics.initStoreMetrics(storesToBackup);
```

Comment:
Why are blob store manager metrics being passed and intialized here instead of in BlobStoreManager init?

Reply:
Same for all other occurences.